packages/remote-feature-flag-controller/src/remote-feature-flag-controller.ts

```diff
@@ -0,0 +1,219 @@
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import { createDeferredPromise } from '@metamask/utils';
+
+import type { AbstractClientConfigApiService } from './client-config-api-service/abstract-client-config-api-service';
+import { projectLogger, createModuleLogger } from './logger';
+import type { FeatureFlags } from './remote-feature-flag-controller-types';
+
+// === GENERAL ===
+
+export const controllerName = 'RemoteFeatureFlagController';
+export const DEFAULT_CACHE_DURATION = 24 * 60 * 60 * 1000; // 1 day
+const log = createModuleLogger(projectLogger, 'ClientConfigApiService');
+
+// === STATE ===
+
+export type RemoteFeatureFlagControllerState = {
+  remoteFeatureFlag: FeatureFlags;
```

Comment:
I see that this was changed recently. Just want to double-check, should this be `remoteFeatureFlags` instead of `remoteFeatureFlag`, since there could be multiple?

``` suggestion
  remoteFeatureFlags: FeatureFlags;
```

Reply:
Done in 8f3c4d5f27ac450768b288b7fbc8d7932d98b7fa